server/src/main/java/org/opensearch/rest/action/cat/RestPitSegmentsAction.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.cat;
+
+import org.opensearch.action.admin.cluster.state.ClusterStateRequest;
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.admin.indices.segments.IndexSegments;
+import org.opensearch.action.admin.indices.segments.IndexShardSegments;
+import org.opensearch.action.admin.indices.segments.IndicesSegmentResponse;
+import org.opensearch.action.admin.indices.segments.PitSegmentsAction;
+import org.opensearch.action.admin.indices.segments.PitSegmentsRequest;
+import org.opensearch.action.admin.indices.segments.ShardSegments;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.common.Table;
+import org.opensearch.common.logging.DeprecationLogger;
+import org.opensearch.index.engine.Segment;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.action.RestActionListener;
+import org.opensearch.rest.action.RestResponseListener;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.rest.RestRequest.Method.GET;
+
+/**
+ * Rest action for pit segments
+ */
+public class RestPitSegmentsAction extends AbstractCatAction {
```

Comment:
I believe one important piece of information (lost in `TransportPitSegmentsAction`) it the `pit_id`s the segment belongs to, what do you think?

Reply:
@bharath-techie wdyt?

Reply:
We could probably create a followup issue to work on that

Reply:
Sorry missed this - will add an issue and work as a followup , we were thinking if we can group "index, shard, prirep, ip, id and pitId" and show results below for this group.

```
-- index, shard, prirep, ip, id , pitId
----- segment, generation, docs.count, docs.deleted, size, size.memory, commited, searchable, version, compound
```